spec/controllers/messages_controller_spec.rb

```diff
@@ -0,0 +1,101 @@
+require 'rails_helper'
+
+RSpec.describe MessagesController, type: :controller do
+
+  let(:user) { FactoryGirl.create :user }
+  let(:admin) { FactoryGirl.create :admin }
+
+  describe "#new" do
+    before { request.session[:user_id] = user.id }
+    it "renders the new template" do
+      get :new
+      expect(response).to render_template(:new)
+    end
+
+    it "instantiates a new message instance variable" do
+      get :new
+      expect(assigns(:message)).to be_a_new(Message)
+    end
+  end
+
+  describe "#create" do
+    before { request.session[:user_id] = user.id }
+    context "with valid parameters" do
+      def valid_message_request
+        post :create, params: { message: { email: "test1234@test.com", content: "aaaaaaaaabbbbbbbbbccccccccccdddddddddddeeeeeeeeeefffffffffff"} }
```

Comment:
Can you use `FactoryGirl` to create the messages? Here you can use http://www.rubydoc.info/github/thoughtbot/factory_girl/FactoryGirl/Syntax/Methods to get a messages attributes.

Do the same thing for all other initial uses of messages.